core/src/main/java/org/apache/metamodel/util/FileResource.java

```diff
@@ -19,16 +19,43 @@
 package org.apache.metamodel.util;
 
 import java.io.File;
+import java.io.FileFilter;
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.OutputStream;
 import java.io.Serializable;
+import java.util.Arrays;
+import java.util.Comparator;
 
 /**
  * {@link File} based {@link Resource} implementation.
  */
 public class FileResource implements Resource, Serializable {
 
+    private class DirectoryInputStream extends AbstractDirectoryInputStream<File> {
```

Comment:
Hmmm... I normally put inline classes and enums as the first elements, but that's a matter of taste. I can move it if you prefer it after static fields.